### ISSUE TITLE
Empty asset inside Entry inside a rich text property

### ISSUE THREAD

**Issue by andersekdahl** [contentful/contentful.net#212]
Hi!
We have a quite deep, nested structure, and right now we have an entry, which has a list of entries and one of those entries has a rich text property, and that rich text property contains an embedded entry. The embedded entry has an asset as a property, but the only thing that is "filled" on the asset is `SystemProperties` and all of the fields are `null`.

The first guess was that the structure was too deep, but we've set `include` to be `5` which is enough to fetch all required entries and assets. If we look at the response from the CDN we see that the asset is included in `included.Assets` with the correct id, and the asset there has all the fields.

One of the entries that gets this behavior has the id `4Cqn9B7T7qhyKG5I8puyAl`, which has a field called `image` that points to the asset `2XMbFK37VWx2mYDXrZnQhj`. And that asset is included in the response below, but the `public Asset Image { get; set; }` on our model only has `SystemProperties` set and all other fields are `null`.

Happy to help investigate this more, but I'm not really sure where to start?

The full CDN response is:
```

{
  "sys": {
    "type": "Array"
  },
  "total": 1,
  "skip": 0,
  "limit": 100,
  "items": [
    {
      "sys": {
        "space": {
          "sys": {
            "type": "Link",
            "linkType": "Space",
            "id": "redacted"
          }
        },
        "id": "4viUznzHTlsXd4SsoFtn2I",
        "type": "Entry",
        "createdAt": "2020-10-08T09:30:39.591Z",
        "updatedAt": "2020-10-15T07:54:13.482Z",
        "environment": {
          "sys": {
            "id": "master",
            "type": "Link",
            "linkType": "Environment"
          }
        },
        "revision": 6,
        "contentType": {
          "sys": {
            "type": "Link",
            "linkType": "ContentType",
            "id": "editorial"
          }
        },
        "locale": "en-US"
      },
      "fields": {
        "name": "About",
        "children": [
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "6AlPi1fgtGm5TOiPyaZfwk"
            }
          }
        ],
        "slug": "about",
        "urls": [
          [
            "4viUznzHTlsXd4SsoFtn2I",
            "/about"
          ]
        ]
      }
    }
  ],
  "includes": {
    "Entry": [
      {
        "sys": {
          "space": {
            "sys": {
              "type": "Link",
              "linkType": "Space",
              "id": "redacted"
            }
          },
          "id": "4Cqn9B7T7qhyKG5I8puyAl",
          "type": "Entry",
          "createdAt": "2020-10-08T20:44:16.107Z",
          "updatedAt": "2020-10-15T07:37:50.553Z",
          "environment": {
            "sys": {
              "id": "master",
              "type": "Link",
              "linkType": "Environment"
            }
          },
          "revision": 7,
          "contentType": {
            "sys": {
              "type": "Link",
              "linkType": "ContentType",
              "id": "imageBlock"
            }
          },
          "locale": "en-US"
        },
        "fields": {
          "image": {
            "sys": {
              "type": "Link",
              "linkType": "Asset",
              "id": "2XMbFK37VWx2mYDXrZnQhj"
            }
          },
          "title": "Matta",
          "showTitle": true,
          "textColor": "Light"
        }
      },
      {
        "sys": {
          "space": {
            "sys": {
              "type": "Link",
              "linkType": "Space",
              "id": "redacted"
            }
          },
          "id": "6AlPi1fgtGm5TOiPyaZfwk",
          "type": "Entry",
          "createdAt": "2020-10-09T19:21:12.564Z",
          "updatedAt": "2020-10-15T07:50:30.658Z",
          "environment": {
            "sys": {
              "id": "master",
              "type": "Link",
              "linkType": "Environment"
            }
          },
          "revision": 60,
          "contentType": {
            "sys": {
              "type": "Link",
              "linkType": "ContentType",
              "id": "scrambledImageBlock"
            }
          },
          "locale": "en-US"
        },
        "fields": {
          "primaryImage": {
            "sys": {
              "type": "Link",
              "linkType": "Asset",
              "id": "5Z4Gg2PaMdbDEroAhhzDiy"
            }
          },
          "primaryText": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "heading-1",
                "content": [
                  {
                    "nodeType": "text",
                    "value": "Hej ",
                    "marks": [],
                    "data": {}
                  },
                  {
                    "nodeType": "text",
                    "value": "hej",
                    "marks": [
                      {
                        "type": "italic"
                      }
                    ],
                    "data": {}
                  }
                ],
                "data": {}
              },
              {
                "nodeType": "paragraph",
                "content": [
                  {
                    "nodeType": "text",
                    "value": "hmm",
                    "marks": [
                      {
                        "type": "italic"
                      }
                    ],
                    "data": {}
                  }
                ],
                "data": {}
              },
              {
                "nodeType": "unordered-list",
                "content": [
                  {
                    "nodeType": "list-item",
                    "content": [
                      {
                        "nodeType": "paragraph",
                        "content": [
                          {
                            "nodeType": "text",
                            "value": "Ul ",
                            "marks": [],
                            "data": {}
                          },
                          {
                            "nodeType": "text",
                            "value": "asdasdsa",
                            "marks": [
                              {
                                "type": "italic"
                              }
                            ],
                            "data": {}
                          }
                        ],
                        "data": {}
                      }
                    ],
                    "data": {}
                  },
                  {
                    "nodeType": "list-item",
                    "content": [
                      {
                        "nodeType": "paragraph",
                        "content": [
                          {
                            "nodeType": "text",
                            "value": "Li",
                            "marks": [],
                            "data": {}
                          }
                        ],
                        "data": {}
                      }
                    ],
                    "data": {}
                  },
                  {
                    "nodeType": "list-item",
                    "content": [
                      {
                        "nodeType": "paragraph",
                        "content": [
                          {
                            "nodeType": "text",
                            "value": "Hm",
                            "marks": [],
                            "data": {}
                          }
                        ],
                        "data": {}
                      }
                    ],
                    "data": {}
                  },
                  {
                    "nodeType": "list-item",
                    "content": [
                      {
                        "nodeType": "paragraph",
                        "content": [
                          {
                            "nodeType": "text",
                            "value": "",
                            "marks": [],
                            "data": {}
                          },
                          {
                            "nodeType": "hyperlink",
                            "content": [
                              {
                                "nodeType": "text",
                                "value": "https://lyko.com/sv",
                                "marks": [],
                                "data": {}
                              }
                            ],
                            "data": {
                              "uri": "https://lyko.com/sv"
                            }
                          },
                          {
                            "nodeType": "text",
                            "value": "",
                            "marks": [],
                            "data": {}
                          }
                        ],
                        "data": {}
                      }
                    ],
                    "data": {}
                  }
                ],
                "data": {}
              },
              {
                "nodeType": "ordered-list",
                "content": [
                  {
                    "nodeType": "list-item",
                    "content": [
                      {
                        "nodeType": "paragraph",
                        "content": [
                          {
                            "nodeType": "text",
                            "value": "Test",
                            "marks": [],
                            "data": {}
                          }
                        ],
                        "data": {}
                      }
                    ],
                    "data": {}
                  },
                  {
                    "nodeType": "list-item",
                    "content": [
                      {
                        "nodeType": "paragraph",
                        "content": [
                          {
                            "nodeType": "text",
                            "value": "",
                            "marks": [],
                            "data": {}
                          },
                          {
                            "nodeType": "hyperlink",
                            "content": [
                              {
                                "nodeType": "text",
                                "value": "Nice detta",
                                "marks": [],
                                "data": {}
                              }
                            ],
                            "data": {
                              "uri": "https://www.nk.se/"
                            }
                          },
                          {
                            "nodeType": "text",
                            "value": "",
                            "marks": [],
                            "data": {}
                          }
                        ],
                        "data": {}
                      }
                    ],
                    "data": {}
                  }
                ],
                "data": {}
              },
              {
                "nodeType": "embedded-entry-block",
                "content": [],
                "data": {
                  "target": {
                    "sys": {
                      "id": "4Cqn9B7T7qhyKG5I8puyAl",
                      "type": "Link",
                      "linkType": "Entry"
                    }
                  }
                }
              },
              {
                "nodeType": "blockquote",
                "content": [
                  {
                    "nodeType": "paragraph",
                    "content": [
                      {
                        "nodeType": "text",
                        "value": "Lorem Ipsum is simply dummy text of the printing and ",
                        "marks": [],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": "typesetting ",
                        "marks": [
                          {
                            "type": "bold"
                          },
                          {
                            "type": "italic"
                          },
                          {
                            "type": "code"
                          },
                          {
                            "type": "underline"
                          }
                        ],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": "industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a ",
                        "marks": [],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": "galley",
                        "marks": [
                          {
                            "type": "bold"
                          },
                          {
                            "type": "italic"
                          }
                        ],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": " of type and ",
                        "marks": [],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": "scrambled",
                        "marks": [
                          {
                            "type": "bold"
                          }
                        ],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": " it to ",
                        "marks": [],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": "make",
                        "marks": [
                          {
                            "type": "underline"
                          }
                        ],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": " a type ",
                        "marks": [],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": "specimen",
                        "marks": [
                          {
                            "type": "code"
                          }
                        ],
                        "data": {}
                      },
                      {
                        "nodeType": "text",
                        "value": " book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ip",
                        "marks": [],
                        "data": {}
                      }
                    ],
                    "data": {}
                  }
                ],
                "data": {}
              },
              {
                "nodeType": "hr",
                "content": [],
                "data": {}
              },
              {
                "nodeType": "paragraph",
                "content": [
                  {
                    "nodeType": "text",
                    "value": "",
                    "marks": [],
                    "data": {}
                  }
                ],
                "data": {}
              }
            ]
          },
          "secondaryImage": {
            "sys": {
              "type": "Link",
              "linkType": "Asset",
              "id": "1QyMMjdHpuYRSJ0VUDe9Gf"
            }
          },
          "bottomImage": {
            "sys": {
              "type": "Link",
              "linkType": "Asset",
              "id": "4upQktQtIUwrOpAkupUFBQ"
            }
          },
          "bottomText": {
            "data": {},
            "content": [
              {
                "data": {},
                "content": [
                  {
                    "data": {},
                    "marks": [
                      {
                        "type": "bold"
                      }
                    ],
                    "value": "Hejsan",
                    "nodeType": "text"
                  }
                ],
                "nodeType": "heading-2"
              },
              {
                "data": {},
                "content": [
                  {
                    "data": {},
                    "marks": [
                      {
                        "type": "bold"
                      }
                    ],
                    "value": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem",
                    "nodeType": "text"
                  }
                ],
                "nodeType": "paragraph"
              }
            ],
            "nodeType": "document"
          }
        }
      }
    ],
    "Asset": [
      {
        "sys": {
          "space": {
            "sys": {
              "type": "Link",
              "linkType": "Space",
              "id": "redacted"
            }
          },
          "id": "1QyMMjdHpuYRSJ0VUDe9Gf",
          "type": "Asset",
          "createdAt": "2020-10-09T19:12:13.860Z",
          "updatedAt": "2020-10-09T19:12:13.860Z",
          "environment": {
            "sys": {
              "id": "master",
              "type": "Link",
              "linkType": "Environment"
            }
          },
          "revision": 1,
          "locale": "en-US"
        },
        "fields": {
          "title": "Mask Group 35",
          "file": {
            "url": "//images.ctfassets.net/redacted/1QyMMjdHpuYRSJ0VUDe9Gf/89c892748edc7110aee8b150286480aa/Mask_Group_35_2x.jpg",
            "details": {
              "size": 630097,
              "image": {
                "width": 720,
                "height": 1924
              }
            },
            "fileName": "Mask Group 35@2x.jpg",
            "contentType": "image/jpeg"
          }
        }
      },
      {
        "sys": {
          "space": {
            "sys": {
              "type": "Link",
              "linkType": "Space",
              "id": "redacted"
            }
          },
          "id": "2XMbFK37VWx2mYDXrZnQhj",
          "type": "Asset",
          "createdAt": "2020-10-08T20:44:07.438Z",
          "updatedAt": "2020-10-08T20:44:07.438Z",
          "environment": {
            "sys": {
              "id": "master",
              "type": "Link",
              "linkType": "Environment"
            }
          },
          "revision": 1,
          "locale": "en-US"
        },
        "fields": {
          "title": "Mask Group 4",
          "file": {
            "url": "//images.ctfassets.net/redacted/2XMbFK37VWx2mYDXrZnQhj/c8db6b4247d6ffc6b4df1444376d8fa4/Mask_Group_4_2x.jpg",
            "details": {
              "size": 930451,
              "image": {
                "width": 1440,
                "height": 1440
              }
            },
            "fileName": "Mask Group 4@2x.jpg",
            "contentType": "image/jpeg"
          }
        }
      },
      {
        "sys": {
          "space": {
            "sys": {
              "type": "Link",
              "linkType": "Space",
              "id": "redacted"
            }
          },
          "id": "4upQktQtIUwrOpAkupUFBQ",
          "type": "Asset",
          "createdAt": "2020-10-09T19:13:00.240Z",
          "updatedAt": "2020-10-09T19:13:00.240Z",
          "environment": {
            "sys": {
              "id": "master",
              "type": "Link",
              "linkType": "Environment"
            }
          },
          "revision": 1,
          "locale": "en-US"
        },
        "fields": {
          "title": "Mask Group 31",
          "file": {
            "url": "//images.ctfassets.net/redacted/4upQktQtIUwrOpAkupUFBQ/fbcf35dd852fb9a1ba60c77da3588e33/Mask_Group_31_2x.jpg",
            "details": {
              "size": 362025,
              "image": {
                "width": 1440,
                "height": 960
              }
            },
            "fileName": "Mask Group 31@2x.jpg",
            "contentType": "image/jpeg"
          }
        }
      },
      {
        "sys": {
          "space": {
            "sys": {
              "type": "Link",
              "linkType": "Space",
              "id": "redacted"
            }
          },
          "id": "5Z4Gg2PaMdbDEroAhhzDiy",
          "type": "Asset",
          "createdAt": "2020-10-09T19:05:45.012Z",
          "updatedAt": "2020-10-09T19:05:45.012Z",
          "environment": {
            "sys": {
              "id": "master",
              "type": "Link",
              "linkType": "Environment"
            }
          },
          "revision": 1,
          "locale": "en-US"
        },
        "fields": {
          "title": "Mask Group 34@2x",
          "file": {
            "url": "//images.ctfassets.net/redacted/5Z4Gg2PaMdbDEroAhhzDiy/81ef398bbd9b35cfc5f44758f239c729/Mask_Group_34_2x.jpg",
            "details": {
              "size": 1064645,
              "image": {
                "width": 1200,
                "height": 1924
              }
            },
            "fileName": "Mask Group 34@2x.jpg",
            "contentType": "image/jpeg"
          }
        }
      }
    ]
  }
}
```

**Comment by Roblinde** [comment]
Hi @andersekdahl 

Could this be the issue: https://robertlinde.se/posts/why-is-my-item-null-contentful-net/

Haven't had the time to have a thorough look, but my first guess would be it's related to what I outlined in that post.

Hope this helps.

**Comment by andersekdahl** [comment]
I don't think so, we've already set that to true because it has solved previous problems for us. 😄

**Comment by Roblinde** [comment]
I see, I'll have a look.

**Comment by Roblinde** [comment]
What object/objects are you trying to deserialize this into? I'll try to replicate this on my end.

**Comment by Roblinde** [comment]
I've found the problem and will release a version with a fix shortly.

**Comment by andersekdahl** [comment]
Lovely, thank you!

**Comment by Roblinde** [comment]
Solved in version 5.0.3
